iceoryx_hoofs/platform/win/source/semaphore.cpp

```diff
@@ -76,7 +80,8 @@ int iox_sem_timedwait(iox_sem_t* sem, const struct timespec* abs_timeout)
     }
 
     time_t epochCurrentTimeDiffInSeconds = abs_timeout->tv_sec - tv.tv_sec;
-    long milliseconds = epochCurrentTimeDiffInSeconds * 1000 + ((abs_timeout->tv_nsec / 1000) - tv.tv_usec) / 1000;
+    long milliseconds = epochCurrentTimeDiffInSeconds * 1000 + (abs_timeout->tv_nsec / 1000000 - tv.tv_usec / 1000);
```

Comment:
What was the reason for this change? The previous one seem to be more accurate?

Reply:
Well, that's not quite correct. In your example the resulting time would be 9.101ms and 9.001001ms but with the new calculation you get 10ms.  Is this intended?
How about
```
auto nanosecondsLeft = abs_timeout->tv_nsec;
auto nanosecondsRight = tv.tv_usec * 1000;
auto rounding = nanosecondsLeft >= nanosecondsRight ? 500000 : -500000;
(nanosecondsLeft - nanosecondsRight + rounding) / 1000000
```
This would correctly round up and down depending if the actual value is above or below .5 and if the result is positive or negative

Reply:
The 10ms are intended, then my example wasn't so good after all.